mqbench/custom_quantizer.py

```diff
@@ -659,3 +659,226 @@ def implicit_merge_patterns(self) -> list:
+                        (node.op == 'call_module' and isinstance(modules[node.target], self._passed_module_type)))
+        for node in nodes:
+    
+            if (node.op == "call_module" and node.target in self.exclude_module_name) or \
+                ((node.op == 'call_function' or node.op == 'call_method') and
+                 node.target in self.exclude_function_type) or \
+                    node.name in self.exclude_node_name:
+                continue
+            if passed_node(node):
+                continue
+            if node.op == 'placeholder':
+                node_need_to_quantize_output.append(node)
+                continue
+            is_output = False
+            # last layer do not quantize
+            for next_node in node.users:
+                if next_node.op == 'output':
+                    is_output = True
+            if is_output:
+                continue
+            for next_node in node.users:
+                if quanlified_node(next_node):
+                    node_need_to_quantize_output.append(node)
+                    break
+        return node_need_to_quantize_output
+
+    def _weight_quant(self, model: GraphModule, qconfig):
+        logger.info("Replace module to qat module.")
+        wqconfig_8bit = copy.deepcopy(qconfig)
+        wq_symmetry = True if is_symmetric_quant(qconfig.weight.p.keywords['qscheme']) else False
+        numbits = 8
+        logger.info('Now all weight quantizers will effectively use only 7 bits out of 8 bits. This resolves the overflow issue problem on AVX2 and AVX-512 machines.')
+        wqconfig_8bit.weight.p.keywords['quant_min'] = -2 ** (numbits - 2) if wq_symmetry else 0
+        wqconfig_8bit.weight.p.keywords['quant_max'] = 2 ** (numbits - 2) - 1 if wq_symmetry else 2 ** (numbits - 1) - 1
+        wqconfig_8bit.weight.p.keywords['factory_kwargs'] = {'not_calc_quant_min_max':True}
+        flattened_qconfig_dict = get_flattened_qconfig_dict({'': wqconfig_8bit})
+        propagate_qconfig_(model, flattened_qconfig_dict)
+        self._qat_swap_modules(model, self.additional_qat_module_mapping)
+        return model
+
+
+    def _insert_fake_quantize_for_act_quant(
+            self,
+            model: GraphModule,
+            qconfig: Any):
+        graph = model.graph
+        modules = dict(model.named_modules())
+        nodes = list(model.graph.nodes)
+
+        quantizer_postfix = "_post_act_fake_quantizer"
+        node_to_quantize_output = self._find_act_quants(model)
+        node_to_quantize_output = OrderedDict.fromkeys(node_to_quantize_output).keys()
+
+        aqconfig_8bit = copy.deepcopy(qconfig.activation)
+        aq_symmetry = True if is_symmetric_quant(qconfig.activation.p.keywords['qscheme']) else False
+        aqconfig_8bit.p.keywords['quant_min'] = 0
+        aqconfig_8bit.p.keywords['quant_max'] = 2 ** 8 - 1
+        aqconfig_8bit.p.keywords['factory_kwargs'] = {'not_calc_quant_min_max':True}
```

Comment:
Run "flake8 ." at the root directory to check the code style.

Reply:
OK，I will resolve these conversion using this tool.